nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/ImagingComponent.java

```diff
@@ -419,10 +416,14 @@ protected PictureView computeView(DocumentModel doc, Blob blob, PictureConversio
         String viewFilename = String.format("%s_%s.%s", title, FilenameUtils.getBaseName(blob.getFilename()),
                 extension);
         viewBlob.setFilename(viewFilename);
+        ImageInfo viewBlobImageInfo = getImageInfo(viewBlob);
         pictureViewMap.put(PictureView.FIELD_FILENAME, viewFilename);
         pictureViewMap.put(PictureView.FIELD_CONTENT, (Serializable) viewBlob);
-        pictureViewMap.put(PictureView.FIELD_INFO, getImageInfo(viewBlob));
-
+        pictureViewMap.put(PictureView.FIELD_INFO, viewBlobImageInfo);
+        if (viewBlobImageInfo != null) {
```

Comment:
since you are going to check for null anyway, why not adding the viewBlobImageInfo only if not null ? It would avoid adding a potentially useless key to the map. Unless you rely on Map#containsKey afterward ?

Reply:
this might slightly change the previous behavior since potential null info was put in the Map: 
```pictureViewMap.put(PictureView.FIELD_INFO, getImageInfo(viewBlob));```